### PR TITLE
DOCSP-30934 release notes

### PR DESCRIPTION
**Pull request by davemungo**
[staging](https://docs-mongodbcom-staging.corp.mongodb.com/cluster-sync/docsworker-xlarge/DOCSP-30934-release-notes-1.4.1/release-notes/1.4/)

[jira](https://jira.mongodb.org/browse/DOCSP-30934)

[build](https://workerpool-boxgs.mongodbstitch.com/pages/job.html?collName=queue&jobId=6494c6d7423952aeb98f45d3)

### CODE REVIEW

**Review by mmcclimon — APPROVED** [review]
(You haven't asked yet, but just to keep the ball rolling on tech review.) This LGTM, thanks for the quick work!

File: source/release-notes/1.4.txt [hunk 1]
```diff
@@ -12,7 +12,23 @@ Release Notes for ``mongosync`` 1.4
...
+~~~~~~~~~~~~~~~~~~~~~
+
+Issues fixed:
+
+- Fixes a build misconfiguration so that telemetry information is sent
+  correctly.
+
+.. _1.4.0-c2c-release-notes:
+
+**Version 1.4.0 released June 21, 2023**
```
Comment (mdb-ashley) [inline source/release-notes/1.4.txt:30]: [question] Does it make sense to have the release versions and dates have the same format? Also, I'm finding it a little difficult to distinguish between the releases visually. Maybe a different heading type and/or some additional space would help.

**Review by mdb-ashley — CHANGES_REQUESTED** [review]
Thanks for picking this up, Dave! It looks good -- I just had one concern about the spacing. LMK when you're ready for me to give it another look.

**Review by davemungo — COMMENTED** [review]
@mdb-ashley here's the page updated to the style we discussed

**Review by mdb-ashley — APPROVED** [review]
LGTM! Thanks, Dave!

**Pull request merged**